src/Ryujinx.Ava/UI/ViewModels/MainWindowViewModel.cs

```diff
@@ -1068,9 +1068,9 @@ private async Task HandleFirmwareInstallation(string filename)
                 {
                     Logger.Error?.Print(LogClass.Application, ex.ToString());
 
-                    static async void Action() => await UserErrorDialog.ShowUserErrorDialog(UserError.NoKeys);
+                    static async Task Action() => await UserErrorDialog.ShowUserErrorDialog(UserError.NoKeys);
 
-                    Dispatcher.UIThread.Post(Action);
+                    await Dispatcher.UIThread.InvokeAsync(Action);
```

Comment:
```suggestion
                    await Dispatcher.UIThread.InvokeAsync(asymc () => await UserErrorDialog.ShowUserErrorDialog(UserError.NoKeys));
```

Everywhere else we inline these, so lets be consistent and do that here as well.

The UIThread isn't used a few lines below, this should probably be changed as well.

A few more lines below `Post()` is used instead of `InvokeAsync()`, did you do that on purpose?